model/host/volume.go

```diff
@@ -124,6 +125,33 @@ func FindVolumesToDelete(expirationTime time.Time) ([]Volume, error) {
 	return findVolumes(q)
 }
 
+// FindVolumesWithTerminatedHost finds volumes that are attached to a host we have marked as terminated, indicating the
+// volume is stuck in an invalid state.
+func FindVolumesWithTerminatedHost() ([]Volume, error) {
+	match := bson.M{
+		"$match": bson.M{
+			VolumeHostKey: bson.M{"$exists": true},
+		}}
+	lookup := bson.M{
+		"$lookup": bson.M{
+			"from":         Collection,
+			"localField":   "host",
+			"foreignField": "_id",
+			"as":           "host_doc",
+		}}
+	matchTerminatedHosts := bson.M{
+		"$match": bson.M{
+			"host_doc.status": evergreen.HostTerminated,
+		}}
+	project := bson.M{"$project": bson.M{"host_doc": 0}}
```

Comment:
It looks like we are not querying on this field rn. Should we add an index?

Reply:
Which field are you referencing Chaya?

Reply:
scratch that, I misread what "host_doc" was.